### PR TITLE
build: on binaryCompatibility & build to 5.3.14

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

File: kafka/build.gradle [hunk 1]
```diff
@@ -2,12 +2,6 @@ plugins {
     id 'io.micronaut.build.internal.kafka-module'
 }
 
-micronautBuild {
```
Comment (sdelamo) [inline kafka/build.gradle:5]: Binary compatibility was turned off https://github.com/micronaut-projects/micronaut-kafka/commit/229bd9b3c1b6c73bc5739440b1fe4ebcac90a4f4

File: gradle.properties [hunk 2]
```diff
@@ -1,6 +1,5 @@
 projectVersion=4.4.0-SNAPSHOT
 projectGroup=io.micronaut.kafka
-micronautSharedSettingVersion=5.3.10
```
Comment (sdelamo) [inline gradle.properties:3]: I have moved this directly to settings to align to the [project template](https://github.com/micronaut-projects/micronaut-project-template/pull/246)

**Review by timyates — APPROVED** [review]
Still not sure why everything in this module is failing on graal 17/latest

https://ge.micronaut.io/s/xwpap35vvdydi/console-log?task=:tests:tasks-sasl-plaintext:nativeCompile

**Comment by graemerocher** [comment]
@timyates let's see if https://github.com/micronaut-projects/micronaut-kafka/pull/579/commits/c13412835f6d7e45a07416b0d2912952994398b8 reveals any more

**Comment by timyates** [comment]
@graemerocher https://github.com/micronaut-projects/micronaut-kafka/pull/581 🤞

File: settings.gradle [hunk 3]
```diff
@@ -1,6 +1,6 @@
 pluginManagement {
     plugins {
-        id 'io.micronaut.build.shared.settings' version getProperty("micronautSharedSettingVersion")
```
Comment (dstepanov) [inline settings.gradle:3]: You can modify the version in gradle properties

**Comment by timyates** [comment]
@sdelamo 💚

File: settings.gradle (hunk 3 above)
Comment (sdelamo) [reply settings.gradle:3]: Yes, I understand. But we should keep the project's build align. Having the version in `settings.gradle`, gets automatic version upgrades PR.

If we want to define build tools version in `gradle.properties`  we should change it in project template and then use it in every project.

**Pull request merged**